nirodents/workflows/brainextraction.py

```diff
@@ -53,121 +59,193 @@ def init_rodent_brain_extraction_wf(
+            (clip_inu, buffernode, [("out", "hires_target")]),
+        ])
+    elif bids_suffix == "t1w":
+        wf.connect([
+            # resampling and laplacian; no truncation or N4
+            (inputnode, res_target, [("in_files", "in_file")]),
+            (res_target, buffernode, [("out_file", "lowres_target")]),
+        ])
+
+    wf.connect([
+        # ants AI inputs
+        (buffernode, init_aff, [("lowres_target", "moving_image")]),
+        (res_tmpl, gauss_tmpl, [("out_file", "in_file")]),
+        (gauss_tmpl, lowres_tmpl, [("out", "in_file")]),
+        (lowres_tmpl, init_aff, [("out_file", "fixed_image")]),
+    ])
+
+    # Graft a template registration-mask if present
+    tpl_regmask_path = get_template(
+        in_template, resolution=2, atlas=None, desc="BrainCerebellumExtraction", suffix="mask",
     )
+    if tpl_regmask_path:
+        lowres_mask = pe.Node(
+            ApplyTransforms(
+                input_image=_pop(tpl_regmask_path),
+                transforms="identity",
+                interpolation="MultiLabel",
+                float=True),
+            name="lowres_mask",
+            mem_gb=1
+        )
 
-    lap_tmpl = pe.Node(ImageMath(operation="Laplacian", op2="0.4"), name="lap_tmpl")
-    if tpl_target_path:
-        lap_tmpl.inputs.op1 = tpl_target_path
-    lap_target = pe.Node(ImageMath(operation="Laplacian", op2="0.4"), name="lap_target")
+        hires_mask = pe.Node(
+            ApplyTransforms(
+                input_image=_pop(tpl_regmask_path),
+                transforms="identity",
+                interpolation="NearestNeighbor",
+                float=True),
+            name="hires_mask",
+            mem_gb=1
+        )
+        wf.connect([
+            (lowres_tmpl, lowres_mask, [("out_file", "reference_image")]),
+            (lowres_mask, init_aff, [("output_image", "fixed_image_mask")]),
+            (res_tmpl, hires_mask, [("out_file", "reference_image")]),
+        ])
 
-    norm_lap_tmpl = pe.Node(ImageMath(operation="Normalize"), name="norm_lap_tmpl")
-    norm_lap_target = pe.Node(ImageMath(operation="Normalize"), name="norm_lap_target")
+    # Spatial normalization step
+    lap_tmpl = pe.Node(ImageMath(operation="Laplacian", op2="0.4 1.0"), name="lap_tmpl")
+    lap_target = pe.Node(ImageMath(operation="Laplacian", op2="0.4 1.0"), name="lap_target")
 
     # Merge image nodes
     mrg_target = pe.Node(niu.Merge(2), name="mrg_target")
     mrg_tmpl = pe.Node(niu.Merge(2), name="mrg_tmpl")
-    # mrg_tmpl.inputs.in1 = tpl_target_path
 
-    # Create integration nodes to allow compatibility between pipelines
-    integrate_1 = pe.Node(niu.IdentityInterface(fields=["in_file"]), name="integrate_1")
-    integrate_2 = pe.Node(niu.IdentityInterface(fields=["in_file"]), name="integrate_2")
+    # norm_lap_tmpl = pe.Node(ImageMath(operation="Normalize"), name="norm_lap_tmpl")
+    # norm_lap_target = pe.Node(ImageMath(operation="Normalize"), name="norm_lap_target")
```

Comment:
I didn't have any luck normalizing the Laplacian with the operation, hence the two-stage process, although I used a scalar (i.e. `0.4 1`). If replacing this with a float improves this then we can probably get rid of these lines